addons/binding/org.openhab.binding.miele/src/main/java/org/openhab/binding/miele/internal/discovery/MieleApplianceDiscoveryService.java

```diff
@@ -0,0 +1,168 @@
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonElement;
+
+/**
+ * The {@link MieleApplianceDiscoveryService} tracks appliances that are
+ * associated with the Miele@Home gateway
+ *
+ * @author Karel Goderis - Initial contribution
+ */
+public class MieleApplianceDiscoveryService extends AbstractDiscoveryService implements ApplianceStatusListener {
+
+    private final Logger logger = LoggerFactory.getLogger(MieleApplianceDiscoveryService.class);
+
+    private final static int SEARCH_TIME = 60;
+
+    private MieleBridgeHandler mieleBridgeHandler;
+
+    public MieleApplianceDiscoveryService(MieleBridgeHandler mieleBridgeHandler) {
+        super(MieleApplianceHandler.SUPPORTED_THING_TYPES, SEARCH_TIME, false);
+        this.mieleBridgeHandler = mieleBridgeHandler;
+    }
+
+    public void activate() {
+        mieleBridgeHandler.registerApplianceStatusListener(this);
+    }
+
+    @Override
+    public void deactivate() {
+        removeOlderResults(new Date().getTime());
+        mieleBridgeHandler.unregisterApplianceStatusListener(this);
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return MieleApplianceHandler.SUPPORTED_THING_TYPES;
+    }
+
+    @Override
+    public void startScan() {
+        List<HomeDevice> appliances = mieleBridgeHandler.getHomeDevices();
+        if (appliances != null) {
+            for (HomeDevice l : appliances) {
+                onApplianceAddedInternal(l);
+            }
+        }
+    }
+
+    @Override
+    protected synchronized void stopScan() {
+        super.stopScan();
+        removeOlderResults(getTimestampOfLastScan());
+    }
+
+    @Override
+    public void onApplianceAdded(HomeDevice appliance) {
+        onApplianceAddedInternal(appliance);
+    }
+
+    private void onApplianceAddedInternal(HomeDevice appliance) {
+        ThingUID thingUID = getThingUID(appliance);
+        if (thingUID != null) {
+            ThingUID bridgeUID = mieleBridgeHandler.getThing().getUID();
+            Map<String, Object> properties = new HashMap<>(2);
+            properties.put(APPLIANCE_ID,
+                    StringUtils.right(appliance.UID, appliance.UID.length() - new String("hdm:ZigBee:").length()));
+            for (JsonElement dc : appliance.DeviceClasses) {
+                if (dc.getAsString().contains("com.miele.xgw3000.gateway.hdm.deviceclasses.Miele")
+                        && !dc.getAsString().equals("com.miele.xgw3000.gateway.hdm.deviceclasses.MieleAppliance")) {
+                    properties.put(DEVICE_CLASS, StringUtils.right(dc.getAsString(), dc.getAsString().length()
+                            - new String("com.miele.xgw3000.gateway.hdm.deviceclasses.Miele").length()));
+                    break;
+                }
+            }
+
+            DiscoveryResult discoveryResult = DiscoveryResultBuilder.create(thingUID).withProperties(properties)
+                    .withBridge(bridgeUID).withLabel(getId(appliance)).build();
```

Comment:
I need to check and test this as i do not remember what Miele puts in the HomeDevice.Name field